tools/crm_rule.c

```diff
@@ -63,24 +63,66 @@ crm_rule_check(pe_working_set_t *data_set, const char *rule_id, crm_time_t *effe
     /* Rules are under the constraints node in the XML, so first find that. */
     cib_constraints = get_object_root(XML_CIB_TAG_CONSTRAINTS, data_set->input);
 
-    /* Get all rules matching the given ID, which also have only a single date_expression
-     * child whose operation is not 'date_spec'.  This is fairly limited, but it's hard
-     * to check expressions more complicated than that.
+    /* Get all rules matching the given ID which are also simple enough for us to check.
+     * For the moment, these rules must only have a single date_expression child and:
+     * - Do not have a date_spec operation, or
+     * - Have a date_spec operation that contains years= but does not contain moon=.
+     *
+     * We do this in steps to provide better error messages.  First, check that there's
+     * any rule with the given ID.
```

Comment:
Let's separate this into two commits, one for breaking the xpath into steps and the other for processing years

I only say this because lately I've been annoyed looking at some of my own past commits and thinking I really should have split them into three or four :)